File: golang/binary_search/guess_number_higher_lower.go
package main

func guess(num int) int {
	target := 6

	if num < target {
		return 1
	} else if num > target {
		return -1
	}

	return 0
}

func guessNumber(n int) int {
	low, high := 1, n

	for low <= high {
		mid := low + (high-low)/2

		result := guess(mid)

		if result == 0 {
			return mid
		} else if result < 0 {
			high = mid - 1
		} else {
			low = mid + 1
		}
	}

	return -1
}

File: java/src/main/java/com/sanedge/interview/binary_search/GuessNumberHigherLower.java
package com.sanedge.interview.binary_search;

public class GuessNumberHigherLower {

    public int guess(int num) {
        int target = 6;

        if (num < target) {
            return 1;
        } else if (num > target) {
            return -1;
        }

        return 0;
    }

    public int guessNumber(int n) {
        int low = 1;
        int high = n;

        while (low <= high) {
            int mid = low + (high - low) / 2;

            int result = guess(mid);

            if (result == 0) {
                return mid;
            } else if (result < 0) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }

        return -1;
    }
}
